enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,179 @@
+| AllRequestBodiesIn | addition to logging metadata for all requests, logs request bodies for every read and write request to the API servers (get, list, create, update, patch). This profile has the most resource overhead.                               |
+| None               | No requests are logged; even OAuth access token requests and OAuth authorize token requests are not logged.                                                                                                                           |
+
+**Audit Log File Storage Location:**
+
+- `path` field to specify non-default audit log file paths.  Accepted values must be absolute paths.
+
+```yaml
+apiServer:
+  auditLog:
+    path: STRING
+```
+
+**Audit Log File Rotation:**
+
+- `maxFileSize` config variable to set maximum audit log file size. Accepted values must be a string with a number and unit suffix with no separator character, e.g. `10GB`. Valid units are "KB", "MB", "GB", "TB", "KiB", "MiB", "GiB", and "TiB".
+
+```yaml
+apiServer:
+  auditLog:
+    maxFileSize: STRING
+```
+
+**Audit Log Retention Policy:**
+
+- `retainPolicy` variable for system response when disk space full
+- Supported values: "ignore", "rotate", "terminate"
+
+| Policy      | Description                                             |
+|-------------|---------------------------------------------------------|
+| `ignore`    | Do nothing.                                             |
+| `rotate`    | Trigger the specified rotation behavior                 |
+| `terminate` | Kill the MicroShift process to prevent loss of log data |
+
+**Logging:**
+
+- Log audit log policy additions, removals, changes
+- Log audit log rotations and locations
+
+### Implementation Details
+
+**Passing User Provided Options to the API Server**
+MicroShift specifies config options to the API server via a file located at `/etc/microshift/resources/kube-apiserver-audit-policies/default.yaml`.  The file is structured as a `KubeAPIServerConfig` custom resource, provided by the `kubecontrolplane.config.openshift.io/v1` API group. Overrides for the specified fields are set in the same directory in a file named `config-overrides.yaml`.  To pass user-defined audit configuration options to the API server, MicroShift will append the user-specified options to `config-overrides.yaml`. This minimizes the complexity of the option-passing and allows for MicroShift defaults to continue to be specified in the `default.yaml` and `config-overrides.yaml` files, as they are now.  This also provides an elegant fallback behavior for user-facing fields. If a user does not specify a value for a field, the default value will be used, just as it is now.  Should a user remove a field they previously set, the field will likewise be removed form the `config-overrides.yaml` file and the default value will no longer be overridden. All audit logging configuration fields must be optional.  If not specified, the existing default audit logging policy will be used.
```

Comment:
The path is in `/var/lib`, not `/etc`.

I only see the `default.yaml` file being written out. It looks like the overrides are part of the assets compiled into MicroShift.

Reply:
You're correct, I've updated the paragraph to describe a different approach that resuses existing logic.